examples/mesos-client-example/src/main/scala/com/mesosphere/mesos/examples/MesosClientExampleFramework.scala

```diff
@@ -43,26 +43,23 @@ class MesosClientExampleFramework(settings: MesosClientSettings) extends StrictL
 
   val client = Await.result(MesosClient(settings, frameworkInfo).runWith(Sink.head), 10.seconds)
 
-  client.mesosSource
-    .runWith(Sink.foreach { event =>
-      if (event.getType == Event.Type.SUBSCRIBED) {
-        logger.info("Successfully subscribed to mesos")
-      } else if (event.getType == Event.Type.OFFERS) {
-
-        val offerIds = event.getOffers.getOffersList.asScala.map(_.getId).toList
-
-        Source(offerIds)
-          .via(info(s"Declining offer with id = ")) // Decline all offers
-          .map(
-            oId =>
-              client.calls.newDecline(
-                offerIds = Seq(oId),
-                filters = Some(Filters.newBuilder().setRefuseSeconds(5.0).build())
-            ))
-          .runWith(client.mesosSink)
-      }
-
-    })
+  client.mesosSource.mapConcat { event =>
+    if (event.getType == Event.Type.SUBSCRIBED) {
+      logger.info("Successfully subscribed to mesos")
+      List.empty
```

Comment:
You could do it like this. However, IMHO the code on the left side, though less optimized, captures the idea of "event loop reacting to events" better than the one on the right side.

Reply:
Kind of. Anyhow, this example did not expose an issue in `outgoingConnection` which is closed when the server sends the `Connection: close` header. The fix will be part of #94.